library/src/scala/quoted/Reporting.scala

```diff
@@ -0,0 +1,35 @@
+package scala.quoted
+
+object Reporting {
```

Comment:
Or, can we make `Reporting` a trait, and make `QuoteContext` extend `Reporting`, the same as it is done in Dotty?

```Scala
trait Reporting { thisCtx: Context =>
  // ...
}
```

For end-users, they don't need to know the concept `Reporting`, which improves usability.

Reply:
That would not help to decouple the concepts. The idea is not to use the QuoteContext directly at all and only let it implicitly track the current scope.

Reply:
> The idea is not to use the QuoteContext directly at all and only let it implicitly track the current scope.

Thakns for explanation, I can see it is a good design principle. The little question still in my mind is why not use `qctx` directly if it can improve usability, like it's done in Dotty?

Reply:
The problem is that based on the users, the usability of this feature was not great. It may look simple to us because that is the way it is done internally in dotty. I prefer to make it clear for normal users than for users that know the compiler internals.